engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/KafkaAvroSourceFactory.scala

```diff
@@ -53,9 +54,14 @@ class KafkaTypedAvroSourceFactory[T: TypeInformation](schemaFactory: Deserializa
              @ParamName("schema")
              @NotBlank
              avroSchema: String): Source[T] with TestDataGenerator = {
-    val schemaRegistryClient = schemaRegistryClientFactory.createSchemaRegistryClient(kafkaConfig)
-    new KafkaSource(consumerGroupId = processMetaData.id, List(topic), schemaFactory.create(List(topic), kafkaConfig),
-      Some(AvroToJsonFormatter(schemaRegistryClient, topic, formatKey)), processObjectDependencies) with ReturningType {
+
+    new KafkaSource(
+      consumerGroupId = processMetaData.id,
+      List(topic),
+      schemaRegistryProvider.deserializationSchemaFactory.create(List(topic), kafkaConfig),
+      Some(schemaRegistryProvider.recordFormatter(topic, formatKey = false)),
```

Comment:
Why you've inlined `formatKey = false`. What about scenario where you have schemas for both keys and values?

Reply:
It's moved to Provider.